app/storage/metadata_parser.py

```diff
@@ -1,105 +1,123 @@
-        self._validator = validator
-        # the function to execute if the value should be auto-generated
-        self._generator = generator
-        # flag to indicate if the value must exist in the jwt token
-        self.mandatory = mandatory
-
-    def validate(self, original_value):
-        if self._validator:
-            # The parser methods throw exceptions on incorrect data
-            self._validator(original_value)
-
-    def generate(self):
-        if self._generator:
-            return self._generator()
-
-        return None
-
-
-metadata_fields = {
-    'user_id': MetadataField(),
-    'ru_ref': MetadataField(),
-    'ru_name': MetadataField(),
-    'eq_id': MetadataField(),
-    'collection_exercise_sid': MetadataField(),
-    'period_id': MetadataField(),
-    'period_str': MetadataField(),
-    'ref_p_start_date': MetadataField(validator=iso_8601_data_parser),
-    'ref_p_end_date': MetadataField(mandatory=False, validator=iso_8601_data_parser),
-    'form_type': MetadataField(),
-    'survey_url': MetadataField(mandatory=False),
-    'return_by': MetadataField(validator=iso_8601_data_parser),
-    'trad_as': MetadataField(mandatory=False),
-    'employment_date': MetadataField(mandatory=False, validator=iso_8601_data_parser),
-    'region_code': MetadataField(mandatory=False),
-    'language_code': MetadataField(mandatory=False),
-    'tx_id': MetadataField(mandatory=False, validator=uuid_4_parser, generator=id_generator),
-    'variant_flags': MetadataField(mandatory=False),
-    'roles': MetadataField(mandatory=False),
-    'case_id': MetadataField(mandatory=False),
-    'case_ref': MetadataField(mandatory=False),
-    'account_service_url': MetadataField(mandatory=False),
+VALIDATORS = {
+    'date': iso_8601_date_parser,
+    'uuid': uuid_4_parser,
+    'string': lambda *args: None,
+    'object': lambda *args: None
 }
 
+MANDATORY_METADATA = {
+    'eq_id': {
+        'validator': 'string'
+    },
+    'form_type': {
+        'validator': 'string'
+    },
+    'ru_ref': {
+        'validator': 'string'
+    },
+    'collection_exercise_sid': {
+        'validator': 'string'
+    },
+    'tx_id': {
+        'validator': 'uuid'
+    }
+}
 
-def parse_metadata(metadata_to_check):
-    parsed = {}
-    try:
-        for key, field in metadata_fields.items():
-            if key in metadata_to_check:
-                attr_value = metadata_to_check[key]
-                field.validate(attr_value)
-                logger.debug('parsing metadata', key=key, value=attr_value)
-            else:
-                logger.debug('generating metadata value', key=key)
-                attr_value = field.generate()
 
-            parsed[key] = attr_value
-    except (RuntimeError, ValueError, TypeError) as e:
-        logger.error('unable to parse metadata', exc_info=e)
-        raise InvalidTokenException('incorrect data in token')
+def parse_runner_claims(claims):
```

Comment:
This function doesn't actually parse anything does it? It just adds the `tx_id` and then calls the validator.

This might be a more appropriate place to call `clean_leading_trailing_spaces`.

Reply:
I should call it in both this and `validate_schema_metadata` then?. Otherwise its assuming that this one will take care of it? And in terms of tests I would have to call both functions.

Reply:
Moved it to session for a reason but can't remember the reason so 👍